crates/env/src/call/call_builder.rs

```diff
@@ -566,6 +702,61 @@ where
     }
 }
 
+impl<E, Args, RetType> CallBuilder<E, Set<CallV2<E>>, Args, RetType>
+where
+    E: Environment,
+{
+    /// Sets the `ref_time_limit` part of the weight limit for the current cross-contract
+    /// call.
+    pub fn ref_time_limit(self, ref_time_limit: Gas) -> Self {
+        let call_type = self.call_type.value();
+        CallBuilder {
+            call_type: Set(CallV2 {
+                ref_time_limit,
+                ..call_type
+            }),
+            ..self
+        }
+    }
+
+    /// Sets the `proof_time_limit` part of the weight limit for the current
+    /// cross-contract call.
```

Comment:
I wonder if we can tell people here that they can ignore setting the proof limit for standalone chains.